src/transmission_manager.rs

```diff
@@ -0,0 +1,341 @@
+    // We don't care about the other transitions here, but this function will still be triggered
+    // because it watches for any change in InfectiousStatusType.
+    if event.current == InfectiousStatusType::Infectious {
+        // Get the number of infection attempts this person will have.
+        let r_0 = context.get_global_property_value(Parameters).unwrap().r_0;
+        #[allow(clippy::cast_possible_truncation)]
+        #[allow(clippy::cast_sign_loss)]
+        let num_infection_attempts =
+            context.sample_distr(TransmissionRng, Poisson::new(r_0).unwrap()) as usize;
+
+        // Start scheduling infection attempt events for this person.
+        // People who have num_infection_attempts = 0 are still passed through this
+        // logic but don't infect anyone. This is so that so that there is only one
+        // place where we need to handle setting people to recovered.
+        schedule_next_infection_attempt(
+            context,
+            event.person_id,
+            num_infection_attempts,
+            // last_infection_time_uniform is relative to when the agent became infectious.
+            // Basically, it tells us the fraction of the agent's time spent infectious that has passed.
+            0.0,
+        );
+    }
+}
+
+/// Schedule the next infection attempt for the transmitter based on the generation
+/// interval and the number of infection attempts remaining.
+fn schedule_next_infection_attempt(
+    context: &mut Context,
+    transmitter_id: PersonId,
+    num_infection_attempts_remaining: usize,
+    last_infection_time_uniform: f64,
+) {
+    // If there are no more infection attempts remaining, set the person to recovered.
+    if num_infection_attempts_remaining == 0 {
+        context.set_person_property(
+            transmitter_id,
+            InfectiousStatus,
+            InfectiousStatusType::Recovered,
+        );
+    } else {
+        // Schedule the next infection attempt.
+        // Get the next infection attempt time.
+        let next_infection_times = get_next_infection_time(context, last_infection_time_uniform);
+
+        // Schedule the infection attempt. This function (a) grabs a contact at the time of the infection event
+        // to ensure that the contacts are current and (b) evaluates whether the transmission event is successful.
+        context.add_plan(
+            context.get_current_time() + next_infection_times.gi,
+            move |context| {
+                infection_attempt(context, transmitter_id)
+                    .expect("Error finding contact in infection attempt");
+                // Schedule the next infection attempt for this infected agent
+                // once the last infection attempt is over.
+                schedule_next_infection_attempt(
+                    context,
+                    transmitter_id,
+                    num_infection_attempts_remaining - 1,
+                    next_infection_times.uniform,
```

Comment:
I think if  you change this to next_infection_times_uniform and make a temporary, you can remove the `move` because these scalar values will have Copy.

Reply:
Sorry, I couldn't get this to work. I may not be understanding your suggestion correctly. Did you mean the following?

```rust
let next_infection_time_uniform = next_infection_times.0;
context.add_plan(context.get_current_time() + next_infection_times.1, |context| {
    schedule_next_infection_attempt(..., next_infection_time_uniform);
}

```